### DISCUSSION TITLE
Return JSON response from Rowy Webhook to Twilio request

### DISCUSSION THREAD

**Discussion by 1pnb** [AntlerVC/firetable#1171] [Support / Q&A] [answered]
Hello,

I need to return a specific value from Firebase table to via Rowy webhook back to Twilio API. I managed to return am XML response and parse on Twilio with 
`{{widgets.getAutoAnsw.parsed.message._fieldsProto.autoanswer_msg.stringValue}}`

Rowy parser code:

```
const basicParser: Parser = async ({ req, res, db, ref, logging }) => {
  const { body } = req;
  logging.log(body);
  logging.log("basicParser started");

  const contacts = await db.collection("phone_numbers").where("number", "==", body.to).get();
  const contact = contacts.docs[0];
  // await contact.ref.update({ autoanswer_msg: "update message!" });

  res.send({ message: contact });
};
```

I can't find a way to return JSON response with a single value from a row. 

I tried 

```
import { Response } from 'express';

const basicParser: Parser = async ({ req, res, db, ref, logging }) => {
  const { body } = req;
  logging.log(body);
  logging.log("basicParser started");

  const contacts = await db.collection("phone_numbers").where("number", "==", body.to).get();
  const contact = contacts.docs[0];
  await contact.ref.update({ autoanswer_msg: "update message!" });

  res.json({ message: contact.data() });
};
```

but get error `Property 'json' does not exist on type '{ send: (v: any) => void; sendStatus: (status: number) => void; }'.(2339)`

Any help appreciated!

**Comment by BeeBombshell** [comment, marked as answer]
Hey @1pnb, it seems like the `res` object you used is not an instance of the `Response` object from the express package, which provides the `json` method for returning JSON responses. To fix the error, you'll need to use the `Response` object you've imported.

## Some quick improvements

- The import statements need to be inside the `basicParser` function body.  Not modifying the function structure is suggested because it might break your function. We have even added some instructions as comments inside the code, such as not writing code outside the function body.

- Also, to use the npm packages, you need to import them via the Common JS `require` imports, as given below:
  ```js
  const { Response } = require('express');
  ```
